lib/dRep/dRepImpl.py

```diff
@@ -80,102 +77,321 @@ def run_dRep(self, ctx, params):
         """
         # ctx is the context object
         # return variables are: output
-        #BEGIN run_dRep
+ 
+        #BEGIN dereplicate
 
-        
-        dprint('ctx:', ctx)
-        dprint('params:', params)
+        dprint('params', run=locals())
 
 
+        # set up globals ds `_globals` for this API-method run
 
+        reset(_globals) # clear all fields but `debug`
+        _globals.update({
+            **self._globals,
+            'workspace_name': params['workspace_name'],
+            'run_dir': os.path.join(self._globals['shared_folder'], str(uuid.uuid4())),
+            'warnings': [],
+            })
+
+        os.mkdir(_globals.run_dir)
+
```

Comment:
I think this is an excessive amount of linebreaks, and I started losing track of what scope I am still in especially with functions nested in functions